packages/vuetify/src/components/VProgressLinear/VProgressLinear.ts

```diff
@@ -42,61 +54,101 @@ export default mixins(Colorable).extend({
     query: Boolean,
+    reactive: Boolean,
+    rounded: Boolean,
+    stream: Boolean,
+    striped: Boolean,
     value: {
       type: [Number, String],
       default: 0
     }
   },
 
+  data () {
+    return {
+      internalLazyValue: this.value || 0
+    }
+  },
+
   computed: {
+    __cachedBackground (): VNode {
+      return this.$createElement('div', this.setBackgroundColor(this.backgroundColor || this.color, {
+        staticClass: 'v-progress-linear__background',
+        style: this.backgroundStyle
+      }))
+    },
+    __cachedBar (): VNode {
+      return this.$createElement(this.computedTransition, [this.__cachedBarType])
+    },
+    __cachedBarType (): VNode {
+      return this.indeterminate ? this.__cachedIndeterminate : this.__cachedDeterminate
+    },
+    __cachedDeterminate (): VNode {
+      return this.$createElement('div', this.setBackgroundColor(this.color, {
+        staticClass: `v-progress-linear__determinate`,
+        style: {
+          width: convertToUnit(this.normalizedValue, '%')
+        }
+      }))
+    },
+    __cachedIndeterminate (): VNode {
+      return this.$createElement('div', {
+        staticClass: 'v-progress-linear__indeterminate',
+        class: {
+          'v-progress-linear__indeterminate--active': this.active
+        }
+      }, [
+        this.genProgressBar('long'),
+        this.genProgressBar('short')
+      ])
+    },
+    __cachedStream (): VNode | null {
+      if (!this.stream) return null
+
+      return this.$createElement('div', this.setTextColor(this.color, {
+        staticClass: 'v-progress-linear__stream',
+        style: {
+          width: convertToUnit(100 - this.normalizedBufer, '%')
+        }
+      }))
+    },
     backgroundStyle (): object {
       const backgroundOpacity = this.backgroundOpacity == null
         ? (this.backgroundColor ? 1 : 0.3)
         : parseFloat(this.backgroundOpacity)
 
       return {
-        height: this.active ? convertToUnit(this.height) : 0,
         opacity: backgroundOpacity,
-        width: `${this.normalizedBufer}%`
+        width: convertToUnit(this.normalizedBufer, '%')
       }
     },
-
-    effectiveWidth (): number {
-      if (!this.normalizedBufer) {
-        return 0
+    classes (): object {
+      return {
+        'v-progress-linear--absolute': this.absolute,
+        'v-progress-linear--fixed': this.fixed,
+        'v-progress-linear--query': this.query,
+        'v-progress-linear--rounded': this.rounded,
+        'v-progress-linear--striped': this.striped,
+        ...this.themeClasses
       }
-
-      return +this.normalizedValue * 100 / +this.normalizedBufer
     },
-
+    computedTransition (): FunctionalComponentOptions {
+      return this.indeterminate ? VFadeTransition : VSlideXTransition
+    },
     normalizedBufer (): number {
-      if (this.bufferValue < 0) {
-        return 0
-      }
-
-      if (this.bufferValue > 100) {
-        return 100
-      }
-
+      if (this.bufferValue < 0) return 0
```

Comment:
Not really a priority, but this could be made into a one-liner

`Math.max(0, Math.min(100, parseFloat(this.bufferValue)))` 

also I think I might have added a `clamp` function to helpers..